### ISSUE TITLE
port forwarding truncates files randomly "Error Content Length Mismatch"

### ISSUE THREAD

**Issue by DeveloperChris** [microsoft/vscode#205782]
Does this issue occur when all extensions are disabled?: **Yes**

- VS Code Version: 
   - 1.86.1/2
- OS Version:  
   - Windows 10 Enterprise (local)
   - Debian 12 (remote)
   - 

Steps to Reproduce:

1. Create a project on a remote server
2. Create port forwarding using vscode eg: 8080 => 8080
3. browse to content in your browser e.g. http://localhost:8080
4. Open multiple pages and watch for the loading spinner in the browser tab (tried both edge and chrome)
5. if the spinner keeps spinning (sometimes page appears fully loaded) open developer tools and wait
6. error pops up after about 1 minute "Error Content Length Mismatch"
 
![image](https://github.com/microsoft/vscode/assets/212806/0d18aeb2-6577-408f-a96c-31f15b356793)

I have spent over 2 days trying to resolve this issue my first thought was a problem with my code. I could find no problems and realised it was even happening on plain html files, but most often when downloading large resources such as JavaScript includes.

Due to the JavaScript issues I thought it may have been an over restrictive configuration on my enterprise controlled PC. I downloaded Wireshark and watched the traffic, it appeared to be truncated before reaching my PC but I am not an expert so I may have read the logs wrong and didn't trust it.

I ngrep'd the output from the server. I proved it was sending the full script to the browser but the browser wasn't getting the full script. it was being truncated in seemingly random spots.

I started to wonder if the issue was vscode so I disabled port forwarding and used putty to recreate the port forwarding.

The issue disappeared and reappeared again when I went back to vscode's port forwarding.

I killed all node processes on the remote server and restarted everything, the issue persisted.

I downgraded vscode to 1.85.2 The problem appeared to be resolved.

vscode auto updated (sheesh) and the problem returned.

I disabled auto updates and downgraded again this time to 1.86.1 the problem continued.

Finally I downgraded again to 1.85.2 the problem has again disappeared and appears to be stable for now. I'll update if that changes

**Version Information**
Version: 1.86.1 (user setup)
Commit: 31c37ee8f63491495ac49e43b8544550fbae4533
Date: 2024-02-07T09:08:20.941Z
Electron: 27.2.3
ElectronBuildId: 26495564
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.19045

**Comment by frozenzia** [comment]
Yeah, I've also got this problem running on osx.

**Version information**
Version: 1.86.2 (Universal)
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:42:13.651Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Darwin arm64 21.6.0

**Comment by alexr00** [comment]
Are you connecting via the Remote SSH extension? If so, what is the value of `remote.SSH.useExecServer`? Does the issue persist if you disable that (if you don't want to upgrade and potentially downgrade again to try this you can install VS Code insiders, which installs side by side)?

**Comment by frozenzia** [comment]
@alexr00 yes, I am using that extension. `remote.SSH.useExecServer` is already off. I didn't re-upgrade yet, just downloaded the 1.86.2 version, but as luck would have it not getting these errors any more.

**Comment by shijithpk** [comment]
am having this issue too. Like Chris, got this while trying to view a project on a remote server through my local browser.

I use the remote-ssh extension. So when i want to view my sveltekit app which i develop on a remote VM, i do a 'npm run dev' and check the app at http://localhost:5173/. 

What is happening is that just half the header of my website shows up and i see this in my browser console `Failed to load resource: net::ERR_CONTENT_LENGTH_MISMATCH`

Here's the details of my VS code build--
Version: 1.86.2
Commit: 903b1e9d8990623e3d7da1df3d33db3e42d80eda
Date: 2024-02-13T19:41:37.860Z
Electron: 27.2.3
ElectronBuildId: 26908389
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Linux x64 5.15.0-94-generic

**Comment by DeveloperChris** [comment]
I am using the remote-ssh V0.107.1 in the 1.85.2 build I don't know if the version number is different on the 1.86.2 build.

My `remote.SSH.useExecServer` is disabled and I have never used the setting before, so it is not to have been a factor in the problem for me.

**Comment by frozenzia** [comment]
> ......I didn't re-upgrade yet, just downloaded the 1.86.2 version, but as luck would have it not getting these errors any more.

Woohoo! New day, new situation. So. Contrary to my colleague's advice, I again had left VSCode open at the end of the workday. Connection to the remote server was disconnected overnight while laptop was in sleep mode or something. Was prompted to re-establish the connection and did so. Immediately had the errors. Checked the `remote.SSH.useExecServer` setting -- wha?! It was actually ON. Turned it off, tried to reload the page that was getting the errors -- STILL getting the errors. I was skeptical though, so shut down VSCode and restarted it, still have that setting OFF, and not getting those errors now. Don't know if that's just because of a restart, but restarting *yesterday* when I didn't know about this setting certainly did not help. I'll keep an eye on it today, see if it starts acting up again or not.

For others' information -- the default value of this setting has changed to TRUE, which is why it was OFF for me when I downgraded to 85.2 -- I'd never used it, so was just seeing the default value.
